hyperparameter_train_network.py

```diff
@@ -41,69 +46,144 @@ def color_cat(s):
-
-block_id_output = Dense(1, activation='relu',name='block_id_output')(network)
-point_output= Dense(2,activation='relu',name='point_output')(network)
-
-
-final_network=Model(inputs=[state_input,words_input,position_input],outputs=[block_id_output,point_output])
-final_network.compile(loss='mean_squared_error',
-        optimizer='Adam',
-        metrics=['accuracy']) 
+    for i in range(len(state)):
+        curr_state = state[i]
+        assert len(curr_state) % 7 == 0
+        for block_offset in range(0, len(curr_state), 7):
+            curr_state[block_offset + 1] = \
+                ord(curr_state[block_offset + 1]) - ord('A') + 1
+            curr_state[block_offset + 3] = \
+                ord(curr_state[block_offset + 3]) - ord('A') + 1
+            curr_state[block_offset + 2] = \
+                color_cat(curr_state[block_offset + 2]) 
+            curr_state[block_offset + 4] = \
+                color_cat(curr_state[block_offset + 4]) 
+
+    return state
+
+class NetworkConfig:
+    MAX_BLOCK_LAYERS = 5
+    MAX_POINT_LAYERS = 5
+    MAX_BOTH_LAYERS = 5
+    NETWORK_SIZES = [16, 32, 64, 128, 254, 512]
+    def __init__(self, num_blocks, lstm_size, block_layers, point_layers, both_layers):
+        self.lstm_size = lstm_size
+        self.block_layers = block_layers
+        self.point_layers = point_layers
+        self.both_layers = both_layers
+        self.num_blocks = num_blocks
+
+    @staticmethod
+    def random_network_config(num_blocks):
+        def create_layer_seq(num_layers):
+            return list(map(lambda x: random.choice(NetworkConfig.NETWORK_SIZES),
+                            [i for i in range(num_layers)]))
+        num_block_layers = random.randint(1, NetworkConfig.MAX_BLOCK_LAYERS)
+        num_point_layers = random.randint(1, NetworkConfig.MAX_BLOCK_LAYERS)
+        num_both_layers = random.randint(1, NetworkConfig.MAX_BLOCK_LAYERS)
+
+        lstm_size = random.choice(NetworkConfig.NETWORK_SIZES)
+        block_layers = create_layer_seq(num_block_layers)
+        point_layers = create_layer_seq(num_point_layers)
+        both_layers = create_layer_seq(num_both_layers)
+
+        return NetworkConfig(num_blocks, lstm_size, block_layers, point_layers, both_layers)
+
+def create_network(num_blocks, MAX_LEN, N_SYMBOLS):
+    config = NetworkConfig.random_network_config(num_blocks)
+
+    state_input = Input(shape=(7*num_blocks,),name='state_input')
+    words_input = Input(shape=(MAX_LEN,N_SYMBOLS), name='words_input')
+    words_lstm=LSTM(config.lstm_size)(words_input)
+    words_lstm = Dense(128, activation='relu')(words_lstm)
+
+    network_input = keras.layers.concatenate([words_lstm, state_input])
+
+    main_network = BuildNet(config.both_layers, network_input)
+    main_output = Dense(config.num_blocks, activation='softmax', name='main_output')(main_network)
+
+    return Model(
+        inputs=[state_input,words_input],
+        outputs=[main_output]
+    )
 
 ##LOAD ALL THE DATA
```

Comment:
Can we put this block in a function or two? I get an uneasy feeling when code is outside of a function...